From ff003996624773555c3d4ec237f91960b0771dcf Mon Sep 17 00:00:00 2001
Message-ID: <ff003996624773555c3d4ec237f91960b0771dcf.1742668052.git.sharlatanus@gmail.com>
In-Reply-To: <cover.1742668052.git.sharlatanus@gmail.com>
References: <cover.1742668052.git.sharlatanus@gmail.com>
From: Sharlatan Hellseher <sharlatanus@gmail.com>
Date: Thu, 20 Mar 2025 22:46:00 +0000
Subject: [PATCH 02/34] gnu: ccfits: Update to 2.7.

* gnu/packages/astronomy.scm (ccfits): Update to 2.7.
[arguments] <configure-flags>: Build shared library and tests.
[inputs]: Add zlib.
[description]: Adjust fill column indentation.

Change-Id: If573eeb889f0c2feed92843978d6cabda22e627b
---
 gnu/packages/astronomy.scm | 19 ++++++++++++-------
 1 file changed, 12 insertions(+), 7 deletions(-)

diff --git a/gnu/packages/astronomy.scm b/gnu/packages/astronomy.scm
index ceeae2ead1..3081d5d33b 100644
--- a/gnu/packages/astronomy.scm
+++ b/gnu/packages/astronomy.scm
@@ -499,24 +499,29 @@ (define-public casacore
 (define-public ccfits
   (package
     (name "ccfits")
-    (version "2.6")
+    (version "2.7")
     (source
      (origin
        (method url-fetch)
        (uri (string-append
-             "https://heasarc.gsfc.nasa.gov/docs/software/fitsio/ccfits/"
+             "https://heasarc.gsfc.nasa.gov/FTP/software/fitsio/ccfits/"
              "CCfits-" version ".tar.gz"))
        (sha256
-        (base32 "04l6na8vr5xadz3rbx62as79x1ch4994vbb625kx0dz5czdkkd1b"))))
+        (base32 "06mhvvdsaqvk3cc309gv6zd4lcxm5q5aialaq0n77gzczv94cdgn"))))
     (build-system cmake-build-system)
-    (inputs (list cfitsio))
+    (arguments
+     (list
+      #:configure-flags
+      #~(list "-DBUILD_SHARED_LIBS=ON"
+              "-DTESTS=ON")))
+    (inputs (list cfitsio zlib))
     (home-page "https://heasarc.gsfc.nasa.gov/docs/software/fitsio/ccfits/")
     (synopsis "C++ interface to the CFITSIO")
     (description
      "CCfits is an object oriented interface to the cfitsio library.  It is
-designed to make the capabilities of cfitsio available to programmers working in
-C++.  It is written in ANSI C++ and implemented using the C++ Standard Library
-with namespaces, exception handling, and member template functions.")
+designed to make the capabilities of cfitsio available to programmers working
+in C++.  It is written in ANSI C++ and implemented using the C++ Standard
+Library with namespaces, exception handling, and member template functions.")
     (license (license:non-copyleft "file://License.txt"
                                    "See License.txt in the distribution."))))
 
-- 
2.48.1

